### PR TITLE
Simplify scripts/background.js code

### PR DESCRIPTION
**Pull request by vbanos**
Remove commented auto archive code.

Refactor right click context menu code.

Refactor callbacks `chrome.windows.onRemoved` and `chrome.tabs.onRemoved`.

Remove 300+ lines of code. Reduce `eslint` errors by 1000+ (total was ~5000).

### CODE REVIEW

File: scripts/background.js [hunk 1]
```diff
@@ -286,86 +342,36 @@ chrome.webRequest.onCompleted.addListener(function(details) {
                     //If show-all Context is true, Create a window which is focussed and create tabs in it
                     if(event2.showall==true){
                       chrome.windows.create({url:chrome.runtime.getURL("doi.html")+"?url="+message.url, width:800, height:800, top:0, left:0, focused:true},function(win){
-                        windowIdtest=win.id;
-                        chrome.windows.onRemoved.addListener(function (win1) {
```
Comment (anishsarangi) [inline scripts/background.js]: Please add chrome.windows.onRemoved.addListener event for `windowIdtest` in line No- 68

Comment (anishsarangi) [reply scripts/background.js]: Please add `var alexa_url="https://archive.org/services/context/alexa?url="+url;` in line number 448. Thank you.

Comment (vbanos) [reply scripts/background.js]: You can view the code in github "Files changed", click on a specific line and add a comment there.

File: scripts/background.js [hunk 2]
```diff
@@ -62,6 +62,62 @@ function rewriteUserAgentHeader(e) {
 
 myNotID=null;
 
+/**
+ * Close window callback
+ */
+chrome.windows.onRemoved.addListener(function (id) {
```
Comment (anishsarangi) [inline scripts/background.js]: Add for `windowIdtest` also.
Thank you

File: scripts/background.js [hunk 3]
```diff
@@ -286,86 +342,36 @@ chrome.webRequest.onCompleted.addListener(function(details) {
...
                       chrome.windows.create({url:chrome.runtime.getURL("doi.html")+"?url="+message.url, width:800, height:800, top:0, left:0, focused:true},function(win){
-                        windowIdtest=win.id;
-                        chrome.windows.onRemoved.addListener(function (win1) {
-                          if(win1==windowIdtest){
-                            windowIdtest=0;
-                          }
-                        });
+                        windowIdtest = win.id;
                       });
                       var alexa_url="https://archive.org/services/context/alexa?url="+url;
```
Comment (anishsarangi) [inline scripts/background.js]: Could declare `alexa_url` globally and used in every where

File: scripts/background.js [hunk 4]
```diff
@@ -286,86 +344,36 @@ chrome.webRequest.onCompleted.addListener(function(details) {
...
                       });
                       chrome.tabs.create({url:chrome.runtime.getURL("tagcloud.html")+"?url="+message.url,'active':false},function(tab){
                         tabId7=tab.id;
-                        chrome.tabs.onRemoved.addListener(function (tabtest) {
-                          if(tabtest==tabId7){
-                            tabId7=0;
-                          }
-                        });
                       });
                       var tweet_bot_url="http://hoaxy.iuni.iu.edu/#query="+open_url+"&sort=mixed&type=Twitter";
```
Comment (anishsarangi) [inline scripts/background.js]: Could declare tweet_bot_url in the global scope and used it everywhere.

File: scripts/background.js (hunk 3 above)
Comment (anishsarangi) [reply scripts/background.js]: I will add a separate PR for this. At this time, It is all good.
Thank you @vbanos .

**Pull request merged**